src/main/scala/com/microsoft/hyperspace/index/sources/default/DefaultFileBasedSource.scala

```diff
@@ -181,14 +188,14 @@ class DefaultFileBasedSource(private val spark: SparkSession) extends FileBasedS
    * Computes the signature using the given [[LogicalRelation]]. This computes a signature of
    * using all the files found in [[PartitioningAwareFileIndex]].
    *
-   * @param logicalRelation Logical relation to compute signature from.
+   * @param logicalPlan Logical plan to compute signature from.
    * @return Signature computed if the given 'logicalRelation' can be processed by this provider.
    *         Otherwise, None.
    */
-  override def signature(logicalRelation: LogicalRelation): Option[String] = {
-    logicalRelation.relation match {
+  override def signature(logicalPlan: LogicalPlan): Option[String] = {
+    getLogicalRelation(logicalPlan).map(_.relation).flatMap {
       case HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, format, _)
-          if isSupportedFileFormat(format) =>
+        if isSupportedFileFormat(format) =>
```

Comment:
I think the previous formatting is correct.

Reply:
👍